### DISCUSSION TITLE
Recommended sharding point?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [MaterializeInc/materialize#9817] [Q&A]
I am wondering at what point you all would recommend sharding to multiple materialize processes while there is no clustering features.

For example, if a topic is doing 10Gb/s, that probably should be handled by it's own shard? Or maybe we have to look at CPU limits first, i.e. we will hit CPU bottlenecks before bandwidth bottlenecks?

Obviously we lose some features sharding, but for inrelated streams I imagine this would be ok.

**Comment by morsapaes** [comment]
We don't really recommend sharding over multiple Materialize instances yourself, since it's non-trivial to handle consistency and other knick-knacks of distributed dataflow execution. If you want to give horizontal scalability a try, though, you can play around with [this prototype](https://github.com/MaterializeInc/materialize/blob/main/test/cluster/README.md)!
